pkg/sql/opt/metadata.go

```diff
@@ -313,20 +312,13 @@ func (md *Metadata) CheckDependencies(
 			return false, nil
 		}
 
-		for privs := md.deps[i].privileges; privs != 0; {
-			// Strip off each privilege bit and make call to CheckPrivilege for it.
-			// Note that priv == 0 can occur when a dependency was added with
-			// privilege.Kind = 0 (e.g. for a table within a view, where the table
-			// privileges do not need to be checked). Ignore the "zero privilege".
-			priv := privilege.Kind(bits.TrailingZeros32(uint32(privs)))
-			if priv != 0 {
+		privs := md.deps[i].privileges
+		for priv := privilege.MinKind; priv <= privilege.MaxKind; priv <<= 1 {
+			if privs&privilegeBitmap(priv) != 0 {
```

Comment:
Can we keep this implemented the same - iterate over the set privileges rather than all possible privileges? Then we don't need `MaxKind` and `MinKind`, AFAICT.